### PR TITLE
Use Guava 20.0-SNAPSHOT and GWT 2.8.0-beta1 to fix the GWT build

### LINKED ISSUE #1323: Switch to Guava 20 for the GWT build
The Travis CI has been broken for a week because Guava 19.0-SNAPSHOT has been removed from google-snapshots. Since the Guava 19.0 release does not work with GWT 2.8.0-beta1, switch to Guava 20.0-SNAPSHOT once that archive is published.

See https://github.com/google/guava/issues/2259.

### PR DESCRIPTION
**Pull request by Dominator008**
Fixes #1323

### CODE REVIEW

**Comment by Dominator008** [comment]
I'm merging this now to fix the Travis CI that has been broken for a week.

**Comment by ChadKillingsworth** [comment]
:+1: After the fact - I was just waiting on Travis to finish.

File: pom-gwt.xml [hunk 1]
```diff
@@ -122,41 +122,16 @@
...
 
     <dependency>
       <groupId>com.google.gwt</groupId>
       <artifactId>gwt-dev</artifactId>
-      <version>2.8.0-SNAPSHOT</version>
+      <version>2.8.0-beta1</version>
     </dependency>
   </dependencies>
 
-  <repositories>
```
Comment (MatrixFrog) [inline pom-gwt.xml]: We still need this section, I would think? Guava 20.0-SNAPSHOT is not published on https://repo1.maven.org/maven2/com/google/guava/guava/ it looks like.

Comment (Dominator008) [inline pom-gwt.xml]: It's published at https://oss.sonatype.org/content/repositories/snapshots/com/google/guava/guava/20.0-SNAPSHOT/

Comment (MatrixFrog) [inline pom-gwt.xml]: Exactly, so why are you deleting the section that points to oss.sonatype.org?

Comment (Dominator008) [inline pom-gwt.xml]: Yeah it's a bit confusing here... The one I deleted points to https://oss.sonatype.org/content/repositories/google-snapshots, which does host Guava.

gwt-2.8.0-beta1 uses oss-parent as the parent POM (see https://repo1.maven.org/maven2/com/google/gwt/gwt/2.8.0-beta1/gwt-2.8.0-beta1.pom):
`
<parent>
<groupId>org.sonatype.oss</groupId>
<artifactId>oss-parent</artifactId>
<version>4</version>
</parent>
`
That POM (https://repo1.maven.org/maven2/org/sonatype/oss/oss-parent/4/oss-parent-4.pom)
adds the https://oss.sonatype.org/content/repositories/snapshots repository, which actually contains Guava 20.0-SNAPSHOT.

So I guess it's fine to delete it here.

**Pull request merged**